flang/include/flang/Runtime/extensions.h

```diff
@@ -22,6 +22,9 @@ extern "C" {
 // CALL FLUSH(n) antedates the Fortran 2003 FLUSH statement.
 void FORTRAN_PROCEDURE_NAME(flush)(const int &unit);
 
+// GNU extension subroutine FDATE
+void FORTRAN_PROCEDURE_NAME(fdate)(std::byte *string, std::int64_t length);
```

Comment:
Why `std::byte` here instead of the usual `char`?

Reply:
it was `std::int8_t` but when I was doing for getlog https://github.com/llvm/llvm-project/pull/74628#discussion_r1420503257, pointed out that it would be better to use std::byte since I have to cast them into `char` anyways.

Reply:
if it is `char` then it won't work for some reason

Reply:
Like what?

Reply:
its my problem, I design the test wrong, it can be `char`, I will change it in the next commit

Reply:
should I change `getlog` and `getarg` as well?